### PR TITLE
Logging in Functionality

### PR DESCRIPTION
**Pull request by abitblue**

### CODE REVIEW

File: interface/api.py [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
+from functools import wraps
+
+from netaddr import EUI
+from login.utils import mutually_exclusive
+
 from interface.wrapper import ResponseData
 from django.utils import timezone
 from typing import Optional, Union
-import requests
+import httpx
```
Comment (harveyj123) [inline interface/api.py:11]: why did you use httpx over requests

File: templates/login/base.html [hunk 2]
```diff
@@ -22,11 +22,13 @@
     {% block content %} {% endblock %}
 
     {% block footer %}
-        <div class="footerStroke"></div>
+        <div class="footerDivider"></div>
         <p class="footerText">Staten Island Technical High School, <a href="{% url 'debug' %}" style="text-decoration: none; color: inherit;">Technology Department</a></p>
         <p class="footerText">Contact: Mr. Whalen, Room C56, or byod@sitechhs.com</p>
-        <p class="footerText">Team: Ryan Yang '20, Ivan Zheng '22, Harvey Jiang '23, Nick Melkadze '20, Anthony Chen</p>
-        <p class="footerText lastFooter">Special Thanks to Daniel Indictor '20, Michael Whalen, and Mark Erlenwein</p>
+        <p class="footerText">Team: Ryan Yang '20, Ivan Zheng '22, Harvey Jiang '23, You Wei '24, Nick Melkadze '20, Anthony Chen</p>
```
Comment (AntoDono) [inline templates/login/base.html]: Can you combine You and Wei so it is Youwei

File: login/views/result.py [hunk 3]
```diff
@@ -16,18 +16,16 @@ def get(self, request: HttpRequest):
...
+    already_registered_template_name = 'login/error_already_registered.html'
+
     # Map from Error to Verbose Name and Possible fixes
     err_map: dict[str, tuple[str, list[str]]] = {
         'unknown': (
             'Unknown Error',
             ['We were unable to process your request.',
              'Please seek assistance in room C56.']
         ),
-        'alreadyRegistered': (
```
Comment (AntoDono) [inline login/views/result.py:26]: Why is this removed?

Comment (AntoDono) [reply login/views/result.py:26]: Why not keep it in the dict for clarity?

File: login/static/style.css [hunk 4]
```diff
@@ -22,51 +38,25 @@ img {
...
-    color: gray;
-    font-size: 1.5vh !important;
 }
 
-.footerStroke {
+.footerDivider {
     margin: 5vh auto auto;
-    width: 60vw;
+    width: 80%;
     border-radius: 1000px;
```
Comment (harveyj123) [inline login/static/style.css:60]: use % or rem

File: interface/api.py (hunk 1 above)
Comment (abitblue) [reply interface/api.py:11]: async support, for whenever we add that

File: login/static/style.css (hunk 4 above)
Comment (abitblue) [reply login/static/style.css:60]: what? this is already in % ...

File: login/static/style.css [hunk 5]
```diff
@@ -188,6 +154,23 @@ select {
...
+    cursor: pointer;
+    outline: none;
+    text-decoration: none;
+    display: inline-block;
+    padding: 0.5vw 0;
+    margin: 1vh 1vw;
+    color: inherit;
+    font-size: 2vh;
+    width: 17vh;
+    border-radius: 1000vh;
```
Comment (harveyj123) [inline login/static/style.css:173]: how does 1000vh even work  ---> %/rem

File: login/static/style.css [hunk 6]
```diff
@@ -204,7 +187,7 @@ select {
     opacity: 0;
     -ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=0)";
     top: 0;
-    right: -20px;
+    right: -2vh;
```
Comment (harveyj123) [inline login/static/style.css:190]: % are better

**Review by harveyj123 — CHANGES_REQUESTED** [review]

File: templates/login/base.html (hunk 2 above)
Comment (IvanZ505) [reply templates/login/base.html]: yes

File: login/views/result.py (hunk 3 above)
Comment (abitblue) [reply login/views/result.py:26]: b/c the default error page results in lots of ppl coming to whalen when they see the "uh oh!" page.
therefore, we moved it outside of the dictionary. though, yes we should make a comment about it.

Comment (abitblue) [reply login/views/result.py:26]: fixed in latest revision.

File: login/static/style.css (hunk 5 above)
Comment (abitblue) [reply login/static/style.css:173]: uhhh, honestly I'm not sure...
this code was grandfathered in.

File: login/static/style.css (hunk 6 above)
Comment (abitblue) [reply login/static/style.css:190]: I think we tried percent, and we couldn't get it to look right, since the text and button scale with vh.

File: templates/login/base.html (hunk 2 above)
Comment (abitblue) [reply templates/login/base.html]: fixed in latest revision

**Review by harveyj123 — APPROVED** [review]

**Comment by AntoDono** [comment]
bro this is like so sus
